### ISSUE TITLE
unix: support long path names in pipe.c

### ISSUE THREAD

**Issue by bnoordhuis** [libuv/libuv#2826]
The BSDs let you pass path names longer than `sizeof(s.sun_path)`, they read past the end when `socklen > sizeof(struct sockaddr_un)` up to a limit<sup>1</sup>. Let's support that.

<sup>1</sup> I'm 80% sure it's 255 on all BSDs but it certainly is on Darwin and FreeBSD.

**Comment by moghya** [comment]
@bnoordhuis I would like to work on this.
I have started looking into pip. `src/unix/pipe.c`. I suppose change is needed [here](https://github.com/libuv/libuv/blob/v1.x/src/unix/pipe.c#L268). Do you have any additional pointers for me ?

**Comment by bnoordhuis** [comment]
@moghya There are more places, grep for `bind` and `connect`.

I haven't thought too deeply about it but I think it calls for a union:
```c
union uv__sockaddr_un {
  struct sockaddr_un sun;
  char extra[offsetof(struct sockaddr_un, sun_path) + 255];
};
```

**Comment by osallak** [comment]
hello @bnoordhuis  i want to work on this issue, could you please assign it to me?

**Comment by bnoordhuis** [comment]
@osallak we don't assign issues (no cookie licking), just go ahead and open a pull request

**Comment by geeksilva97** [comment]
https://github.com/libuv/libuv/pull/2917#issuecomment-1686173174 I have mac. I want to try this out.

Is this still issue relevant, @bnoordhuis ?

**Comment by bnoordhuis** [comment]
@geeksilva97 yes, still relevant; you can use my old PR as a starting point if you want

**Comment by geeksilva97** [comment]
> [@geeksilva97](https://github.com/geeksilva97) yes, still relevant; you can use my old PR as a starting point if you want

Great! Will do!

**Comment by osallak** [comment]
@geeksilva97 any updates? If you have not finished it yet, I would like to work on it.

**Comment by osallak** [comment]
I will try to finish it, thanks. Could you also let me know how you are doing? Any advice on how to go about doing this? Where should I start?
